### PR TITLE
Reconstruction of TxnPool.

### PR DESCRIPTION
**Pull request by luodanwg**
Reconstruction of TxnPool

1.package transaction pool.
  all transaction should flow into txnpool first, when txnpool's
  validation passed, it will add into the process transaction scope.
2.implement of transaction pool validation.
  Flowchart:https://www.processon.com/view/link/5939550be4b05a4d43805ffc

Tests:
I have test with 20W reg transaction has about 150s in my own computer.
(before fix is about 140s)and have test with
reg->issue->transfer->record all seems ok.

Signed-off-by: luodanwg <luodan.wg@gmail.com>

### CODE REVIEW

**Comment by luodanwg** [comment]
fixed.

File: net/node/transactionPool.go [hunk 1]
```diff
@@ -5,98 +5,310 @@ import (
...
 	"fmt"
 	"sync"
 )
 
 type TXNPool struct {
 	sync.RWMutex
-	txnCnt uint64
-	list   map[common.Uint256]*transaction.Transaction
+	txnCnt        uint64                                      // count
+	txList        map[common.Uint256]*transaction.Transaction // transaction which have been verifyed will put into this map
```
Comment (dreamfly281) [inline net/node/transactionPool.go]: here all the transaction abbreviation is txn instead of tx to avoid mess up with transmit abbreviation tx.

Comment (luodanwg) [inline net/node/transactionPool.go]: thanks, I have rename it to txnList ,could you helps to check it again, thanks.

File: net/node/transactionPool.go [hunk 2]
```diff
@@ -5,98 +5,312 @@ import (
...
+}
+
+func (this *TXNPool) incrAssetIssueAmountSummary(assetId common.Uint256, delta common.Fixed64) {
+	this.Lock()
+	defer this.Unlock()
+	this.issueSummary[assetId] = this.issueSummary[assetId] + delta
+}
+
+func (this *TXNPool) decrAssetIssueAmountSummary(assetId common.Uint256, delta common.Fixed64) {
+	this.Lock()
```
Comment (dreamfly281) [inline net/node/transactionPool.go]: Did the transaction pool need record all the issue amount?

**Pull request merged**